python/revng/pypeline/container.py

```diff
@@ -150,38 +152,61 @@ def contains_all(self, obj: ObjectSet) -> bool:
         return self.objects().issubset(obj)
 
 
+def loads_container(
+    container_type: type[Container],
+    data: str,
+) -> Container:
+    container = container_type()
+    obj_id_ty = get_singleton(ObjectID)  # type: ignore[type-abstract]
+    container.deserialize(
+        {
+            obj_id_ty.deserialize(obj_id): (
+                content if container.is_text() else base64.b64decode(content)
+            )
+            for obj_id, content in json.loads(data).items()
+        }
+    )
+    return container
+
+
 def load_container(
     container_type: type[Container],
-    path: str,
+    path: str | Path,
 ) -> Container:
     """
     Load a container from a serialized format.
     This is used to **load** cached objects into this container.
     """
-    container = container_type()
-    obj_id_type = get_singleton(ObjectID)  # type: ignore[type-abstract]
     with open(path, "r", encoding="utf-8") as f:
-        container.deserialize(
-            {obj_id_type.deserialize(k): bytes.fromhex(v) for k, v in json.load(f).items()}
-        )
-    return container
+        return loads_container(container_type, f.read())
+
+
+def dumps_container(
```

Comment:
1. Why is this not a method of container?
2. The `dumps` and `loads` convention is rather poor, let's not follow it.
   `.to_string()` and `.from_string()`?